extensions-core/s3-extensions/src/main/java/org/apache/druid/storage/s3/S3TaskLogs.java

```diff
@@ -95,9 +96,15 @@ private Optional<InputStream> streamTaskFile(final long offset, String taskKey)
         }
 
         final GetObjectRequest request = new GetObjectRequest(config.getS3Bucket(), taskKey)
-            .withMatchingETagConstraint("\"" + objectMetadata.getETag() + "\"")
+            .withMatchingETagConstraint(objectMetadata.getETag())
             .withRange(start, end);
 
+        if (objectMetadata.getETag() != null) {
+          if (!objectMetadata.getETag().startsWith("\"") && !objectMetadata.getETag().endsWith("\"")) {
+            request.setMatchingETagConstraints(Collections.singletonList("\"" + objectMetadata.getETag() + "\""));
+          }
+        }
```

Comment:
We can put these lines to a helper method to check and add quotation marks for etag so that the code will be simplied as


```java
final GetObjectRequest request = new GetObjectRequest(config.getS3Bucket(), taskKey)
            .withMatchingETagConstraint(ensureQuotated(objectMetadata.getETag()))
            .withRange(start, end);

static String ensureQuotated(String eTag) {
   ...
}
```

And this help method can also be used by the test cases.

For the test cases, we also need a test case that accept a etag without quotation and check the the etag in the final reuqest has quotation marks.

Reply:
Thanks for the review.

I have added a helper method and test cases as you mentioned.

> For the test cases, we also need a test case that accept a etag without quotation and check the the etag in the final reuqest has quotation marks.

I may not be able to meet your expectations regarding the above test case.
In that case, I would like to modify it and would be glad if you could provide the test pattern for my reference.